dev/io.openliberty.org.testcontainers/src/io/openliberty/org/testcontainers/generate/Dockerfile.java

```
@@ -68,7 +68,7 @@ private static DockerImageName constructImageName(Path location) {
         // Find version (between the last two separator characters)
         int end = fullPath.lastIndexOf(File.separator);
         int start = fullPath.substring(0, end).lastIndexOf(File.separator) + 1;
-        final String version = fullPath.substring(start, end);
+        final String version = fullPath.substring(start-1, end+1); //Include file seperators in the version string since version can be a whole number and that may appear earlier in the path
 
         // Find repository (between "resources/" and version)
         start = fullPath.lastIndexOf("resources/") + 10;
```

```suggestion
        end = start - 1; //End where the version started (exclude the path separator)
        start = fullPath.lastIndexOf("resources/") + 10;
```

NOTE: github won't let me suggest an edit to line 75 so you'll have to manually remove that if you accept this suggestion.